front/src/routes/integration/all/xiaomi/DevicePanel.jsx

```diff
@@ -57,10 +58,10 @@ const DevicePanel = ({ children, ...props }) => (
                   deleteDevice={props.deleteDevice}
                 />
               ))}
-            {props.xiaomiDevices && props.xiaomiDevices.length === 0 && (
-              <p class="text-center">
+            {(!props.xiaomiDevices || props.xiaomiDevices.length === 0) && (
```

Comment:
Why did you change the condition ? 
Not sure this is the goal of the empty state, empty state != loading state

Reply:
The empty state should be displayed only when the state is indeed empty, not loading

Reply:
The loading state should display the loader div inside the dimmer block.
This part is should be displayed only once the getXiaomiDevicesStatus !== Getting.